### ISSUE TITLE
"[jekyll-webmention_io] Liquid error: internal" error when using includes in Webmention templates

### ISSUE THREAD

**Issue by nhoizey** [aarongustafson/jekyll-webmention_io#83]
I have created a few templates for specific Webmention types:
https://github.com/nhoizey/nicolas-hoizey.com/tree/master/_includes/webmentions

The code for the author avatar is copy/pasted multiple times, so I wanted to share it in an include.

But when I try to serve or build the site, I get this error:

```
[jekyll-webmention_io] Liquid error: internal
```

I have to say that I'm currently using version 2.9.1 of this plugin because of incompatible shared dependencies. I'll try to remove the other plugin to test with latest version.

**Comment by nhoizey** [comment]
I made a branch with these includes:
https://github.com/nhoizey/nicolas-hoizey.com/tree/shared-webmention-author-photo/_includes/webmentions

**Comment by aarongustafson** [comment]
Ok weird. I’ll take a look.

Sounds like I need to figure out how to handle includes in includes ;-)

**Comment by twhite96** [comment]
@nhoizey how did you get the plugin to work with your templates? Whenever I build the template tags in the `head` of my blog all have `display:none` as either an id or class. Not sure what I am doing wrong.

**Comment by aarongustafson** [comment]
I am hitting my head against the wall here. It looks like the Liquid parser needs to know about the other Liquid Tags in the project, which I can get from the `site_payload`, but I’m not sure how exactly to merge that info and the @data hash in order to get them both resolved into the template. Any help here would be greatly appreciated :-)

**Comment by ashmaroli** [comment]
@nhoizey Is there any additional info available when you try to build with the above includes via the
`--trace` option..?
```
bundle exec jekyll build --trace
```

**Comment by aarongustafson** [comment]
Worth noting, I tried this with additional liquid tags and get the same error. I tried both `highlight` and `link`. Liquid internals like conditionals are not a problem. It leads me to believe the Liquid parser doesn’t have access to the Jekyll instance tags.

**Comment by aarongustafson** [comment]
Yeah, `raw` works fine.

**Comment by ashmaroli** [comment]
> I tried this with additional liquid tags and get the same error.

Can you provide me with a minimal, reproducible repo that I can use to encounter this error at my end..?
Alternatively, is there any difference in the output if you set `Liquid` to parse `@template` with
`error_mode => :warn`..?

**Comment by aarongustafson** [comment]
I have a local test site I use. It’s a skeleton demo but should work. I don’t have it in Github because there’s no point, but here’s a ZIP of everything but the vendor directory.

[test-site.zip](https://github.com/aarongustafson/jekyll-webmention_io/files/2349903/test-site.zip)

**Comment by ashmaroli** [comment]
I've traced the bug behind this.. will send a PR shortly..

**Comment by nhoizey** [comment]
@twhite96 my build works with my current master branch, but not the one I linked to in the comment before yours.

**Comment by aarongustafson** [comment]
Resolved in [2.9.7](https://github.com/aarongustafson/jekyll-webmention_io/releases/tag/v2.9.7).
